java/flight/flight-grpc/src/main/java/org/apache/arrow/flight/FlightGrpcUtils.java

```diff
@@ -19,17 +19,67 @@
 
 import java.util.Collections;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
 
 import org.apache.arrow.flight.auth.ServerAuthHandler;
 import org.apache.arrow.memory.BufferAllocator;
 
 import io.grpc.BindableService;
+import io.grpc.CallOptions;
+import io.grpc.ClientCall;
 import io.grpc.ManagedChannel;
+import io.grpc.MethodDescriptor;
 
 /**
  * Exposes Flight GRPC service & client.
  */
 public class FlightGrpcUtils {
+  /**
+   * Proxy class for ManagedChannel that makes closure a no-op.
+   */
+  private static class ProxyManagedChannel extends ManagedChannel {
+    private final ManagedChannel channel;
+
+    ProxyManagedChannel(ManagedChannel channel) {
+      this.channel = channel;
+    }
+
+    @Override
+    public ManagedChannel shutdown() {
+      return this;
```

Comment:
This logic seems wrong. We want the ProxyManagedChannel to have proper shutdown mechanics while leaving the wrapped channel untouched.

So we should track if someone has called shutdown(), then change isShutdown(), isTerminated(), newCall, etc accordingly.

newCall should block new API calls if the ProxyManagedChannel has been shutdown (presumably this means throw, not no-op).